helm/vitalam-node/templates/service.yaml

```diff
@@ -46,36 +45,39 @@ spec:
       name: prometheus
 {{- end }}
 ---
-{{- if $.Values.node.perNodeServices.createP2pNodePortService }}
+{{- if $.Values.node.perNodeServices.createP2pService }}
 apiVersion: v1
 kind: Service
 metadata:
   name: {{ $fullname }}-{{ $i }}-rc-p2p
 spec:
-  type: NodePort
-  externalTrafficPolicy: Local
+  {{ include "vitalam-node.serviceType" $ | nindent 2 }}
   selector:
     {{- $selectorLabels | nindent 4 }}
     statefulset.kubernetes.io/pod-name: {{ $fullname }}-{{ $i }}
   ports:
     - name: p2p
       port: 30333
       targetPort: 30333
+  {{- if and $.Values.node.collator.isParachain (or (eq $.Values.node.perNodeServices.p2pServiceType "LoadBalancer") (eq $.Values.node.perNodeServices.p2pServiceType "ClusterIP")) }}
```

Comment:
this is easy to read and understand 🤨 

are variables possible here?

Reply:
Variables - no
Nested If statements but that is even uglier.

Helm does allow this template to be split over multiple lines however the linter blows up.  I'll need to investigate this at a later date.